cern_access/indico_cern_access/templates/regform_identity_data_section.html

```diff
@@ -0,0 +1,71 @@
+{% from 'message_box.html' import message_box %}
+{% from '_switch.html' import switch %}
+
+{% set access_request = registration.cern_access_request %}
+
+{% if not registration or not access_request or (not access_request.has_identity_info and not access_request.is_active) %}
+    <div class="personal-data-form">
+        <div class="action-box highlight" style="margin-top: 10px;">
+            <div class="section">
+                <div class="icon icon-location"></div>
+                <div class="text">
+                    <div class="label">
+                        {%- trans %}Request access to the CERN site{% endtrans -%}
+                    </div>
+                    <div>
+                    </div>
+                </div>
+                <div class="toolbar">
+                    {{ switch(id='toggle-personal-data-form', checked=false, name='ext__request_cern_access') }}
+                </div>
+            </div>
+        </div>
+        <div class="i-box regform-section">
+            <div class="i-box-header">
+                <div class="i-box-title">{% trans %}CERN Access{% endtrans %}</div>
+                <div class="i-box-header-text">
+                    {% trans dates=render_access_dates(start_dt, end_dt) -%}
+                        CERN's Security Policy requires you to provide the following
+                        additional data. The contents of this form will not be shared
+                        with the event organizers and will only be revealed upon request
+                        from CERN's Site Surveillance service.
+                        <br>
+                        <br>
+                        Once you have provided this data and the access has been granted, you will be able to download
+                        a ticket which grants you access to the CERN site during this
+                        period: {{ dates }}
+                    {%- endtrans %}
+                </div>
+            </div>
+            <div class="i-box-content">
+                <div class="i-form">
+                    {{ form_rows(form, orientation='horizontal', fields=('ext__birth_date', 'ext__nationality', 'ext__birth_place')) }}
+                </div>
+            </div>
+        </div>
+    </div>
+
+    <script>
+        (function() {
+            'use strict';
+
+            $(document).on('registration-form-loaded', function() {
+                var registrationForm = $('#registrationForm'),
+                    personalDataForm = $('.personal-data-form').detach();
+
+                registrationForm.find('.regform-section-list').append(personalDataForm);
+            });
+
+            $(document).ready(function() {
+                var personalDataForm = $('.personal-data-form').find('.i-box');
+
+                personalDataForm.hide();
+                $('#toggle-personal-data-form').on('change', function() {
+                    personalDataForm.toggle();
```

Comment:
that's just asking for things to get out of sync. it's better to toggle it to the `this.checked` status and then trigger the change event handler afterwards to have the correct state before the first real change on the field.